pkg/endpoint/endpoint.go

```diff
@@ -278,6 +280,13 @@ type Endpoint struct {
 	DatapathConfiguration models.EndpointDatapathConfiguration
 }
 
+// UpdateAnnotations updates the annotations for this Endpoint.
+func (e *Endpoint) UpdateAnnotations(anno map[string]string) {
+	e.unconditionalLock()
+	defer e.unlock()
+	e.annotations = anno
```

Comment:
I would likely plan to do some additional (simple) processing here on the annotations, to generate something closer to what the policy subsystem can deal with (ie `L4Policy` / `L4PolicyMap`).